dialogue-core/src/main/java/com/palantir/dialogue/core/Channels.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.core;
+
+import static java.util.stream.Collectors.toList;
+
+import com.palantir.dialogue.Channel;
+import java.util.Collection;
+import java.util.List;
+
+public final class Channels {
+
+    private Channels() {}
+
+    public static Channel create(Collection<? extends Channel> channels) {
+        List<LimitedChannel> limitedChannels = channels.stream()
+                .map(ConcurrencyLimitedChannel::create)
+                .collect(toList());
+
+        return new RetryingChannel(new QueuedChannel(new RoundRobinChannel(limitedChannels)));
```

Comment:
it'd kind of be nice if this were something that was flattenable

Reply:
Yeah, going to revisit a builder pattern or something of that sort. The issue right now is that none of these things have the same constructor signatures:

RetryingChannel takes a Channel
QueuedChannel takes a LimitedChannel
RoundRobinChannel takes a List<LimitedChannel>